core-blocks/file/editable-link.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+
+export default class FileBlockEditableLink extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.copyLinkToClipboard = this.copyLinkToClipboard.bind( this );
+		this.showPlaceholderIfEmptyString = this.showPlaceholderIfEmptyString.bind( this );
+
+		this.state = {
+			showPlaceholder: ! this.props.text,
+		};
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.text !== this.props.text ) {
+			this.setState( { showPlaceholder: ! this.props.text } );
+		}
+	}
+
+	copyLinkToClipboard( event ) {
+		const selectedText = document.getSelection().toString();
+		const htmlLink = `<a href="${ this.props.href }">${ selectedText }</a>`;
+		event.clipboardData.setData( 'text/plain', selectedText );
+		event.clipboardData.setData( 'text/html', htmlLink );
+	}
+
+	forcePlainTextPaste( event ) {
+		event.preventDefault();
+
+		const selection = document.getSelection();
+		const clipboard = event.clipboardData.getData( 'text/plain' ).replace( /[\n\r]/g, '' );
+		const textNode = document.createTextNode( clipboard );
+
+		selection.getRangeAt( 0 ).insertNode( textNode );
+		selection.collapseToEnd();
+	}
+
+	showPlaceholderIfEmptyString( event ) {
+		this.setState( { showPlaceholder: event.target.innerText === '' } );
+	}
+
+	render() {
+		const { className, placeholder, text, href, updateFileName } = this.props;
+		const { showPlaceholder } = this.state;
+
+		return (
+			<Fragment>
+				<a
+					aria-label={ placeholder }
+					className={ `${ className }__textlink` }
+					href={ href }
+					onBlur={ ( event ) => updateFileName( event.target.innerText ) }
+					onInput={ this.showPlaceholderIfEmptyString }
+					onCopy={ this.copyLinkToClipboard }
+					onCut={ this.copyLinkToClipboard }
+					onPaste={ this.forcePlainTextPaste }
+					contentEditable
```

Comment:
Warning:

>Warning: A component is `contentEditable` and contains `children` managed by React. It is now your responsibility to guarantee that none of those nodes are unexpectedly modified or duplicated. This is probably not intentional.

Should this be using the `RichText` component instead? There are many complexities of `contentEditable` that are meant to be abstracted into this component for general usage like this.